cmd/mt-parrot/monitor.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/grafana/metrictank/stacktest/graphite"
+	"github.com/grafana/metrictank/stats"
+	log "github.com/sirupsen/logrus"
+	"math"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+type seriesStats struct {
+	lastTs uint32
+	//the partition currently being checked
+	partition int32
+	//the number of nans present in the time series
+	nans int32
+	//the sum of abs(value - ts) across the time series
+	deltaSum float64
+	//the number of timestamps where value != ts
+	numNonMatching int32
+
+	//tracks the last seen non-NaN time stamp (useful for lag
+	lastSeen uint32
+}
+
+func monitor() {
+	for tick := range time.NewTicker(queryInterval).C {
+
+		query := graphite.ExecuteRenderQuery(buildRequest(tick))
+		if query.HTTPErr != nil {
+			stats.NewCounter32("parrot.monitoring.error;error=http").Inc()
```

Comment:
should initialize stats once and keep reusing them

Reply:
0d9c84cf62717a5ce05399aca6818618d6fa70ec